server/project/project.go

```diff
@@ -57,11 +62,61 @@ func (s *Server) Get(ctx context.Context, q *ProjectQuery) (*v1alpha1.AppProject
 	return s.appclientset.ArgoprojV1alpha1().AppProjects(s.ns).Get(q.Name, metav1.GetOptions{})
 }
 
+func getRemovedDestination(oldProj, newProj *v1alpha1.AppProject) map[string]v1alpha1.ApplicationDestination {
+	oldDest := make(map[string]v1alpha1.ApplicationDestination)
+	newDest := make(map[string]v1alpha1.ApplicationDestination)
+	for i := range oldProj.Spec.Destinations {
+		dest := oldProj.Spec.Destinations[i]
+		oldDest[fmt.Sprintf("%s/%s", dest.Server, dest.Namespace)] = dest
+	}
+	for i := range newProj.Spec.Destinations {
+		dest := newProj.Spec.Destinations[i]
+		newDest[fmt.Sprintf("%s/%s", dest.Server, dest.Namespace)] = dest
+	}
+
+	removed := make(map[string]v1alpha1.ApplicationDestination, 0)
+	for key, dest := range oldDest {
+		if _, ok := newDest[key]; !ok {
+			removed[key] = dest
+		}
+	}
+	return removed
+}
+
 // Update updates a project
 func (s *Server) Update(ctx context.Context, q *ProjectUpdateRequest) (*v1alpha1.AppProject, error) {
 	if !s.enf.EnforceClaims(ctx.Value("claims"), "projects", "update", fmt.Sprintf("*/%s", q.Project.Name)) {
```

Comment:
We currently use `*/*` in our rbac rules (i.e. `<project>/<appname>`). At this point, having the `/` only makes sense for applications because all applications belong to projects. Should we use `*/*` only for apps, and `*` for clusters, projects? 

Second: I feel that repositories need to be associated with projects as well (and can follow a `<project>/<repourl>`). Someone should not be able to delete repo credentials to a project that they don't have access to.